static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+import bean from 'bean';
+import reqwest from 'reqwest';
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+
+const addUpdatingState = buttonEl => {
```

Comment:
Could you please flow-type this file a bit (with at least return types)?

Reply:
I don't think we should polish this module any further. I generally avoid dealing with the [weirdness](https://github.com/guardian/frontend/wiki/So-you-want-to-ES6%3F#%EF%B8%8F-weirdness), because I think the risk of conversion and the benefit of doing so is negative especially for a "short-lived" module. I don't think any of our modern technical mechanisms will alleviate the burden for the developer who does eventually have to understand _why_ this module exists (the _what_ question is readable from the current state, in my view). If anything, this module doesn't need types, it just needs comments. I did however, test the module locally works, and each function has parity. I think for es6 functional parity should be priority.